crates/cast/src/lib.rs

```diff
@@ -299,12 +298,22 @@ pub fn get_rpc_error_message(error: StarknetError) -> &'static str {
     }
 }
 
-pub fn handle_rpc_error<T, G>(
-    error: ProviderError<JsonRpcClientError<T>>,
-) -> std::result::Result<G, Error> {
+pub fn handle_rpc_error<T>(error: ProviderError) -> std::result::Result<T, Error> {
     match error {
-        Other(RpcError(Code(error))) => Err(anyhow!(get_rpc_error_message(error))),
-        Other(RpcError(Unknown(error))) => Err(anyhow!(error.message)),
+        Other(x) => {
+            if let Some(err) = x
+                .deref()
+                .as_any()
+                .downcast_ref::<starknet::providers::jsonrpc::RpcError>()
+            {
+                match err {
+                    Code(error) => Err(anyhow!(get_rpc_error_message(error))),
+                    Unknown(error) => Err(anyhow!(error.message.clone())),
+                }
+            } else {
+                Err(anyhow!("Unknown RPC error"))
+            }
+        }
```

Comment:
this parsing is untested, so i have no idea if it will work in production, this looks good to me but i have no idea that in fact this is how the downcasting should go - @THenry14 can you verify?

Reply:
This looks fine to me as well. Seems all the cases are covered and the tests are working, so we should be fine. 

I have also commited fixes to multicall tests that were failing because of scarb version bump - hope that's okay 🙏